pancancer_evaluation/prediction/classification.py

```diff
@@ -163,10 +165,65 @@ def train_lasso(X_train,
                 lasso_penalty,
                 n_folds=5,
                 max_iter=1000,
-                use_sgd=False):
+                use_sgd=False,
+                sgd_lr_schedule='optimal'):
 
     import pancancer_evaluation.config as cfg
+
+    from sklearn.model_selection import train_test_split
+
+    subtrain_ixs, valid_ixs = train_test_split(
+        np.arange(X_train.shape[0]),
+        test_size=(1 / n_folds),
+        shuffle=True
+    )
+    X_subtrain, X_valid = X_train.iloc[subtrain_ixs, :], X_train.iloc[valid_ixs, :]
+    y_subtrain, y_valid = y_train.iloc[subtrain_ixs, :], y_train.iloc[valid_ixs, :]
+
     if use_sgd or cfg.lasso_sgd:
+
+        def get_eta0(lr_schedule):
```

Comment:
I typically do it when I have a short/simple function that I only need to use in a particular scope, since the nested function won't be usable outside of the function containing it. I mostly just do it for readabillity - in this case I found it a bit syntactically easier to read than just putting the if/else block where I need it.

I do think this code has gotten a bit more complicated than I would like, though. I'll think about ways it could be improved.